llvm/lib/IR/Verifier.cpp

```diff
@@ -2231,6 +2231,26 @@ void Verifier::verifyFunctionAttrs(FunctionType *FT, AttributeList Attrs,
   checkUnsignedBaseTenFuncAttr(Attrs, "patchable-function-prefix", V);
   checkUnsignedBaseTenFuncAttr(Attrs, "patchable-function-entry", V);
   checkUnsignedBaseTenFuncAttr(Attrs, "warn-stack-size", V);
+
+  if (auto A = Attrs.getFnAttr("sign-return-address"); A.isValid()) {
+    StringRef S = A.getValueAsString();
+    if (S != "none" && S != "all" && S != "non-leaf")
+      CheckFailed("invalid value for 'sign-return-address' attribute: " + S, V);
+  }
+
+  if (auto A = Attrs.getFnAttr("sign-return-address-key"); A.isValid()) {
+    StringRef S = A.getValueAsString();
+    if (!S.equals_insensitive("a_key") && !S.equals_insensitive("b_key"))
```

Comment:
should this really be case insensitive?

Reply:
This matches the assertion in `AArch64MachineFunctionInfo.cpp`. Although clang/LLVM are inconsistent, some places use `==` some `equals_insensitive`.

Reply:
I intend to change all the places to expect lower case only, for the three branch protection attributes.